Makefile

```diff
@@ -1,5 +1,14 @@
 default: test
 
+.PHONY: setup
+setup:
+	cd tools && GO111MODULE=on go install github.com/golangci/golangci-lint/cmd/golangci-lint
+	cd tools && go mod tidy
+
+.PHONY: lint
+lint:
+	golangci-lint run ./...
```

Comment:
Is there any cleaner way to support local linting? What are the best practices? Would like to avoid having to install local tools if possible, and it would also be nice to avoid the additional go.mod. Linting locally in a Docker process might be a viable alternative?

Reply:
you might be able to use a build version of the linting package.
https://github.com/golangci/golangci-lint/blob/master/build/Dockerfile they have a container with that?